src/main/java/com/fasterxml/jackson/core/json/UTF8JsonGenerator.java

```diff
@@ -1914,6 +1916,18 @@ private final void _writeUTF8Segment2(final byte[] utf8, int offset, int len)
         _outputTail = outputPtr;
     }
 
+    private int[] _extendOutputEscapesTo8Bits() {
+        final int[] escapes = _outputEscapes;
```

Comment:
Ok: I am fine with the idea, but will propose one change: instead of overwriting `_outputEscapes`, let's introduce secondary field (`_outputEscapes8Bit` or whatever), left as `null`, constructed first needed.
Just changes it to null check.

The reason is just that ideally `_outputEscapes` is left as-is for other use cases, to minimize any risk of regression.

Reply:
The benefit of overwriting the field is that it will automatically be recreated if it is reset to a 7-bit wide LUT. Storing the 8-bit table in a separate field might get out of date, especially since `_outputEscapes` is a `protected` field that can be assigned through subclasses (that is evil, of course, but still a possibility).